programs/nft-voter/src/instructions/relinquish_nft_vote.rs

```diff
@@ -0,0 +1,111 @@
+    #[account(mut)]
+    pub beneficiary: UncheckedAccount<'info>,
+}
+
+pub fn relinquish_nft_vote(ctx: Context<RelinquishNftVote>) -> Result<()> {
+    let registrar = &ctx.accounts.registrar;
+
+    // Ensure the Governance belongs to Registrar.realm and is owned by Registrar.governance_program_id
+    let _governance = governance::get_governance_data_for_realm(
+        &registrar.governance_program_id,
+        &ctx.accounts.governance,
+        &registrar.realm,
+    )?;
+
+    // Ensure the Proposal belongs to Governance from Registrar.realm and Registrar.governing_token_mint and is owned by Registrar.governance_program_id
+    let proposal = proposal::get_proposal_data_for_governance_and_governing_mint(
+        &registrar.governance_program_id,
+        &ctx.accounts.proposal,
+        &ctx.accounts.governance.key(),
+        &registrar.governing_token_mint,
+    )?;
+
+    // If the Proposal is still in Voting state then we can only Relinquish the NFT votes if the Vote was withdrawn in spl-gov first
+    // When vote is withdrawn in spl-gov then VoteRecord is disposed and we have to assert it doesn't exist 
+    //
+    // If the Proposal is in any other state then we can dispose NftVoteRecords without any additional Proposal checks
+    if proposal.state == ProposalState::Voting {
+        let vote_record_info = &ctx.accounts.vote_record.to_account_info();
+
+        // Ensure the given VoteRecord address matches the expected PDA
+        let vote_record_key = get_vote_record_address(
+            &registrar.governance_program_id,
+            &registrar.realm, 
+            &registrar.governing_token_mint,
+            &ctx.accounts.governing_token_owner.key(),
+            &ctx.accounts.proposal.key());
+        
+        require!(
+            vote_record_key == vote_record_info.key(),
+            NftVoterError::InvalidVoteRecordForNftVoteRecord
+        );
+
+        require!(
+            // VoteRecord doesn't exist if data is empty or account_type is 0 when the account was disposed in the same Tx
+            vote_record_info.data_is_empty() || vote_record_info.try_borrow_data().unwrap()[0] == 0,
+            NftVoterError::VoteRecordMustBeWithdrawn
+        );
+    }
+
+    // Dispose all NftVoteRecords
+    for nft_vote_record_info in ctx.remaining_accounts.iter() {
+        // Ensure NftVoteRecord is for the given Proposal and TokenOwner
+        let _nft_vote_record = get_nft_vote_record_data_for_proposal_and_token_owner(
+            nft_vote_record_info,
+            &ctx.accounts.proposal.key(),
+            &ctx.accounts.governing_token_owner.key(),
+        )?;
+
+        dispose_account(nft_vote_record_info, &ctx.accounts.beneficiary);
+    }
+
+    let voter_weight_record = &mut ctx.accounts.voter_weight_record;
+
+    // Reset VoterWeightRecord and set expiry to expired to prevent it from being used
+    voter_weight_record.voter_weight = 0;
+    voter_weight_record.voter_weight_expiry = Some(0);
```

Comment:
TIL that Anchor automatically serializes back at the end for you, neat!

Reply:
Yeah, that's nice